src/System.Management.Automation/help/CommandHelpProvider.cs

```diff
@@ -530,6 +539,44 @@ private string GetHelpFile(string helpFile, CmdletInfo cmdletInfo)
 
             string location = MUIFileSearcher.LocateFile(helpFileToLoad, searchPaths);
 
+            // try home location
+            if(String.IsNullOrEmpty(location))
```

Comment:
I believe the code here can be integrated into the `if .. else if ... else` block in the `if (!File.Exists(helpFileToLoad))`.

The code looks like this:
```c#
if (!File.Exists(helpFileToLoad))
{
    helpFileToLoad = Path.GetFileName(helpFileToLoad);

    if (mshSnapInInfo != null)
    {
        Diagnostics.Assert(!string.IsNullOrEmpty(mshSnapInInfo.ApplicationBase),
            "Application Base is null or empty.");
        // not minishell case..
        // we have to search only in the application base for a mshsnapin...
        // if you create an absolute path for helpfile, then MUIFileSearcher
        // will look only in that path.
        searchPaths.Add(mshSnapInInfo.ApplicationBase);
        searchPaths.Add(HelpUtils.GetUserHomeHelpSearchPath());
    }
    else if (cmdletInfo.Module != null && !string.IsNullOrEmpty(cmdletInfo.Module.Path))
    {
        searchPaths.Add(cmdletInfo.Module.ModuleBase);
        // calulate the module folder path in UserHomeHelpSearchPath with the code below.
        searchPaths.Add(calulatedUserSearchModulePath);
    }
    else
    {
        searchPaths.Add(GetDefaultShellSearchPath());
        searchPaths.Add(GetCmdletAssemblyPath(cmdletInfo));
        searchPaths.Add(HelpUtils.GetUserHomeHelpSearchPath());
    }
}
else
{
    helpFileToLoad = Path.GetFullPath(helpFileToLoad);
}

string location = MUIFileSearcher.LocateFile(helpFileToLoad, searchPaths);
```

The current code passes in `helpFileToLoad` in case that `cmdletInfo.mshSnapInInfo` or `cmdletInfo.module` is available. This is not necessary because if you look at `MUIFileSearcher.LocateFile`, even if `helpFileToLoad` is a full file path, it still gets first normalized to `filename` and `parent folder path`, and then search the `filename` in the folders.

Reply:
Fixed.